### PR TITLE
Service Registration: Variable

### PR DESCRIPTION
**Pull request by kmh287**
This pull request distinguishes installing/registering the variable service and creating it.Please see #7839 for more context.

/to @choumx @jridgewell

### CODE REVIEW

File: extensions/amp-analytics/0.1/variables.js [hunk 1]
```diff
@@ -302,8 +302,19 @@ export class VariableService {
 
 /**
  * @param {!Window} win
+ */
+export function installVariableService(win) {
+  registerServiceBuilder(win,
+                         'amp-analytics-variables',
+                         VariableService,
+                         /* opt_factory */ undefined);
```
Comment (jridgewell) [inline extensions/amp-analytics/0.1/variables.js]: Unnecessary param?

File: extensions/amp-analytics/0.1/variables.js [hunk 2]
```diff
@@ -302,8 +302,19 @@ export class VariableService {
...
+}
+
+/**
+ * @param {!Window} win
  * @return {!VariableService}
  */
 export function variableServiceFor(win) {
-  return fromClass(win, 'amp-analytics-variables', VariableService);
+  installVariableService(win);
```
Comment (choumx) [inline extensions/amp-analytics/0.1/variables.js]: Installing here may not be necessary if you add `installVariableService` calls to test-amp-analytics.js and test-variables.js.

Comment (kmh287) [inline extensions/amp-analytics/0.1/variables.js]: done

File: extensions/amp-analytics/0.1/variables.js [hunk 3]
```diff
@@ -302,8 +302,17 @@ export class VariableService {
 
 /**
  * @param {!Window} win
+ */
+export function installVariableService(win) {
+  registerServiceBuilder(win,
+                         'amp-analytics-variables',
+                         VariableService);
```
Comment (choumx) [inline extensions/amp-analytics/0.1/variables.js]: Nit: Generally use +4 spaces when *invoking* functions. Align with the open parens (or +4 spaces) in declarations. Anyways, I think this fits on one line.

Comment (kmh287) [inline extensions/amp-analytics/0.1/variables.js]: sure

**Pull request merged**